src/legacy/core_plugins/kibana/public/discover/embeddable/search_embeddable_factory.ts

```diff
@@ -20,17 +20,30 @@
 import 'ui/doc_table';
 
 import { EmbeddableFactory } from 'ui/embeddable';
+import {
+  EmbeddableInstanceConfiguration,
+  OnEmbeddableStateChanged,
+} from 'ui/embeddable/embeddable_factory';
+import { SavedSearch, SavedSearchLoader } from '../types';
 import { SearchEmbeddable } from './search_embeddable';
 
 export class SearchEmbeddableFactory extends EmbeddableFactory {
-  constructor($compile, $rootScope, searchLoader) {
+  private $compile: ng.ICompileService;
+  private $rootScope: ng.IRootScopeService;
+  private searchLoader: SavedSearchLoader;
+
+  constructor(
+    $compile: ng.ICompileService,
```

Comment:
You can shorten that in TypeScript, by just writing:

```ts
constructor(
  private $compile: ng.ICompileSerice,
  private $rootScope: ng.IRootScopeService,
  private searchLoader: SavedSearchLoader
)
```

Then you can clean line 31-33 and 41-43 completely. It will automatically make this a private variable and assign the value to it.